src/ape/managers/chain.py

```diff
@@ -182,8 +182,11 @@ def range(
         elif start_or_stop < 0:
             raise ValueError(f"stop '{stop}' cannot be negative.")
 
-        for i in range(start, stop, step):
-            yield self._get_block(i)
+        # Note: the range stop block is a non-inclusive stop.
+        #       Where as the query method is an inclusive stop.
+        results = self.query("*", start_block=start, stop_block=stop - 1)  # type: ignore
+        for _, row in results.iterrows():
+            yield BlockAPI.parse_obj(row.to_dict())
```

Comment:
Shouldn't this be `self.provider.network.ecosystem.decode_block()`? I believe this will have abstract methods on it and not work too great otherwise

Reply:
Yes, please do this!

Reply:
I'm not sure what you mean by this. Should we be doing this:
```python
yield self.provider.network.ecosystem.decode_block()
```
?

Reply:
Yes, call that method with `row.to_dict()`